src/Workspaces/Core/Portable/Workspace/Solution/AdditionalTextWithState.cs

```diff
@@ -14,11 +15,18 @@ namespace Microsoft.CodeAnalysis.Diagnostics
     internal sealed class AdditionalTextWithState : AdditionalText
     {
         private readonly TextDocumentState _documentState;
+        private static readonly ConditionalWeakTable<TextDocumentState, AdditionalText> _additionalTextsForDocumentStates = new();
```

Comment:
See individual commit comment on this -- this is a bit of a hack to keep this PR size down; I suspect the alternate approach will do more churn than the rest of this PR, so doing one at a time may be best.

Reply:
➡️ #54084

Reply:
Looks like my hunch about churn may have not been as bad as I thought, but you're on the right track with the other PR I think.